src/sciline/_provider.py

```diff
@@ -175,8 +175,13 @@ def from_function(cls, provider: ToProvider) -> ArgSpec:
         """Parse the argument spec of a provider."""
         hints = get_type_hints(provider)
         signature = inspect.getfullargspec(provider)
-        args = {name: hints[name] for name in signature.args}
-        kwargs = {name: hints[name] for name in signature.kwonlyargs}
+        try:
+            args = {name: hints[name] for name in signature.args}
+            kwargs = {name: hints[name] for name in signature.kwonlyargs}
```

Comment:
So why did this not raise before? This looks the same to me (apart from a better exception message)?